blockly-rtc/server/db_utils.js

```diff
@@ -45,22 +53,31 @@ function queryDatabase(serverId) {
 };
 
 /**
- * Add rows to the database.
- * Rows are added transactionally.
- * @param {!Promise} Promise object represents the success of the write.
+ * Add an entry to the database.
+ * @param {!LocalEntry} entry The entry to be added to the database.
+ * @return {!Promise} Promise object with the serverId for the entry if the
+ * write succeeded.
  * @public
  */
 function addToServer(entry) {
 
   return new Promise((resolve, reject) => {
-    db.run('INSERT INTO eventsdb(events, entryId) VALUES(?,?)',
-        [JSON.stringify(entry.events), entry.entryId], (err) => {
-      if (err) {
-        console.error(err.message);
-        reject(err);
-      };
+    db.serialize(() => {
+      db.run('INSERT INTO eventsdb(events, entryId) VALUES(?,?)',
+          [JSON.stringify(entry.events), entry.entryId], (err) => {
+        if (err) {
+          console.error(err.message);
+          reject(err);
+        };
```

Comment:
What happens if the second SQL query completes before the first? You resolve before having completed the ``INSERT``

Reply:
What is the relationship between the two queries, should they be chained?

Reply:
Yes, the two are executing sequentially since they are wrapped by db.serialize. More info on db.serialize here: https://www.sqlitetutorial.net/sqlite-nodejs/statements-control-flow/. Based on this, I do not believe the second query can resolve before the first is completed. Let me know if this is incorrect.

Reply:
Yep, that makes sense. Thanks for clarifying.